multi_partner_learning.py

```diff
@@ -435,42 +480,57 @@ def aggregate_model_weights(self):
             new_weights.append(list(avg_weights_for_layer))
 
         return new_weights
+    
 
     def build_model_from_weights(self, new_weights):
         """Generate a new model initialized with weights passed as arguments"""
 
         new_model = self.generate_new_model()
         new_model.set_weights(new_weights)
-        new_model.compile(
-            loss=keras.losses.categorical_crossentropy,
-            optimizer="adam",
-            metrics=["accuracy"],
-        )
 
         return new_model
-
-    def init_with_new_models(self):
+    
+    
+    def init_with_models(self, use_saved_weights=False):
```

Comment:
`use_saved_weights` is an attribute, you could get it without passing it as an argument here, couldn't you?

Reply:
Right!